src/OrleansRuntime/Scheduler/InvokeWorkItem.cs

```diff
@@ -43,7 +43,8 @@ public override void Execute()
             try
             {
                 var grain = activation.GrainInstance;
-                var runtimeClient = (ISiloRuntimeClient)grain.GrainReference.RuntimeClient;
+                var runtime = (GrainReferenceRuntime)grain.GrainReference.Runtime;
```

Comment:
This down-cast is a bit concerning, Should we have some kind of local 'InvokeOnActivation' method instead? Maybe it would require too big of a change for this PR - too many classes to expose or abstract away, what do you think?